pkg/controller/core/clusterqueue_controller.go

```diff
@@ -498,3 +499,17 @@ func (r *ClusterQueueReconciler) updateCqStatusIfChanged(
 	}
 	return nil
 }
+
+func (r *ClusterQueueReconciler) getWorkloadsStatus() *kueue.ClusterQueuePendingWorkloadsStatus {
+	pendingWorkloads := make([]kueue.ClusterQueuePendingWorkload, 0)
+	for _, workloads := range r.qManager.GetSnapshots() {
+		pendingWorkloads = append(pendingWorkloads, workloads...)
+	}
+	if len(pendingWorkloads) == 0 {
+		return nil
+	}
+	return &kueue.ClusterQueuePendingWorkloadsStatus{
+		Head:           pendingWorkloads,
+		LastChangeTime: metav1.Time{Time: time.Now()},
```

Comment:
Let's update the `ClusterQueuePendingWorkloadsStatus` only if its contests changed, in order to save unnecessary updates that just bump the time. We may compare the lists of workloads with `equality.Semantic.DeepEqual`

Reply:
Actually, we should not update the status if the last status was updated less than `UpdateStatusIntervalSeconds`. Still, we may use `equality.Semantic.DeepEqual` to prevent unnecessary changes, but only as a later check.